### PR TITLE
Enable AppVeyor CI

### PR DESCRIPTION
**Pull request by EwoutH**
I create AppVeyor configuration based on rav1e, using cargo to build and test to Windows targets. I also added a Cargo build & test to Travis to exclude Travis specific errors.

Regardless of Cargo or Make is used, crav1e builds seem to fail at the moment. Even with nothing changed the original Travis configuration fails on the current master. 

Travis (`make examples && ./simple_encoding`, which triggers `cargo build --release`)
```
   Compiling crav1e v0.1.0 (/home/travis/build/EwoutH/crav1e)
error: failed to run custom build command for `crav1e v0.1.0 (/home/travis/build/EwoutH/crav1e)`
process didn't exit successfully: `/home/travis/build/EwoutH/crav1e/target/release/build/crav1e-a615b6c52fbdcf1b/build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseSyntaxError { crate_name: "rav1e", src_path: "", error: ParseError(Some("failed to parse crate: failed to parse all tokens")) }', src/libcore/result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

make: *** [target/release/librav1e.a] Error 101
```
AppVeyor (`cargo build --release`)
```
   Compiling crav1e v0.1.0 (C:\projects\crav1e)
error: failed to run custom build command for `crav1e v0.1.0 (C:\projects\crav1e)`
process didn't exit successfully: `C:\projects\crav1e\target\release\build\crav1e-9e08a716a4395d30\build-script-build` (exit code: 101)
--- stderr
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseSyntaxError { crate_name: "rav1e", src_path: "", error: ParseError(Some("failed to parse crate: failed to parse all tokens")) }', src\libcore\result.rs:997:5
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
Command exited with code 101
```

### CODE REVIEW

**Comment by lu-zero** [comment]
Let see what changed meanwhile.... Looks like some dependency broke

**Comment by lu-zero** [comment]
can you reproduce the issue locally?

**Comment by lu-zero** [comment]
found the problem: https://github.com/eqrion/cbindgen/issues/268

I'm quite tempted to consider to work around it by adopting the approach used by servo, even if it is a little overkill.

**Comment by EwoutH** [comment]
Nice find! Feel free to make any changes you like.

**Comment by lu-zero** [comment]
If you could split the travis changes from appveyor I'd merge them right now, I know they are good even if the current bindings generator is having problems.

**Comment by lu-zero** [comment]
for now, could you pin the compiler version to `nightly-2018-12-15` ?

**Comment by EwoutH** [comment]
You probably have to enable [AppVeyor](https://ci.appveyor.com/project/EwoutH/crav1e) somewhere in the GitHub settings to let it show up in the UI.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - appveyor DownloadFile https://www.nasm.us/pub/nasm/releasebuilds/2.14.02/win64/nasm-2.14.02-win64.zip -FileName nasm.zip
+    - 7z e -y nasm.zip
+    - rustup-init -yv --default-toolchain %channel% --default-host %host%
+    - set PATH=%PATH%;%USERPROFILE%\.cargo\bin;%APPVEYOR_BUILD_FOLDER%
+    - rustc -vV
+    - cargo -vV
+    - rustup target add %target%
+
+build_script:
+    - git submodule update --init
```
Comment (lu-zero) [inline appveyor.yml]: crav1e doesn't have submodules.

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - cargo build --release --target=%target%
+    
+# Temporary search to find artifacts
+after_build:
+    - dir *.exe /s /b | find ^"temp^" /v /i | findstr /e .exe
```
Comment (lu-zero) [inline appveyor.yml]: crav1e right now should just produce a library.

Comment (EwoutH) [reply appveyor.yml]: Any idea how the libary is called/named?

Comment (lu-zero) [reply appveyor.yml]: either it is `librav1e.lib` or `librav1e.a`, but I never tried crav1e on windows...

**Comment by lu-zero** [comment]
Great :)

**Pull request merged**